### PR TITLE
Minor performance fixes to debouncing logic 

### PR DESCRIPTION
**Pull request by Rikimbili**

### CODE REVIEW

File: web/src/pages/feed.tsx [hunk 1]
```diff
@@ -43,20 +45,12 @@ export default function Feed({ searchValue }: Props) {
 
   // Debounce the search value (don't immediately update it after every keystroke)
   const debouncedSetSearchValue = useMemo(
-    () =>
-      debounce(() => {
-        return void refetch();
-      }, 250),
+    () => debounce(setDebouncedSearchValue, 250),
     []
   );
```
Comment (rob-3) [inline web/src/pages/feed.tsx:50]: I'm pretty sure this useMemo is not worthwhile

Comment (rob-3) [reply web/src/pages/feed.tsx:50]: It only prevents the closure from being recreated and adds additional reading complexity that isn't worth it

File: web/src/pages/feed.tsx [hunk 2]
```diff
@@ -67,6 +61,13 @@ export default function Feed({ searchValue }: Props) {
     });
   }, [hasNextPage, isFetchingNextPage]);
 
+  // Cleanup and cancel subscriptions
+  useEffect(() => {
+    return () => {
+      debouncedSetSearchValue.cancel();
+    };
+  }, []);
```
Comment (rob-3) [inline web/src/pages/feed.tsx:69]: I'm nervous about the empty dependency array here because I think you really want `[debouncedSetSearchValue]`

**Review by rob-3 — CHANGES_REQUESTED** [review]

**Comment by rob-3** [comment]
I think the eslint warning do have a point here, and at worst we just do a little extra computation.

File: web/src/pages/feed.tsx (hunk 1 above)
Comment (Rikimbili) [reply web/src/pages/feed.tsx:50]: I'm just gonna take a hooks approach and implement debouncing via a useDebounce hook

Comment (rob-3) [reply web/src/pages/feed.tsx:50]: I think what you're doing here is fine, but one of us is confused about what `useMemo()` does

Comment (rob-3) [reply web/src/pages/feed.tsx:50]: I'm pretty sure the main intention is to cache creation of large arrays or data structures when their dependencies don't change, which is not what is happening here.

Comment (Rikimbili) [reply web/src/pages/feed.tsx:50]: This is probably not the best useMemo use case, useCallback might be good too or even useRef. The main reason of why I went with this is that I saw an article pointing out that the lodash debounce function is expensive and should only be initialized or ran once when the component mounts. I didn't do any performance benchmark to test this tho and this approach requires too much setup. So a simple useDebounce hook might just be what we want

Comment (rob-3) [reply web/src/pages/feed.tsx:50]: Yeah then `useCallback` is the right thing I think

**Review by rob-3 — APPROVED** [review]

**Pull request merged**